File: src/engsoft/jogo/patos/Pato_Borracha.java
package engsoft.jogo.patos;

public class Pato_Borracha extends Pato {

	public Pato_Borracha()
	{
		setComportamento(new Nao_Voa(), new Nao_Pula());	
	}
	
	public String mostrar() {
		return "Ola, eu sou de Boarracha.";
	}
	
}

File: Pato_Borracha.ts
import { Pato } from "./Pato";
import { Nao_Voa } from "./Nao_Voa";
import { Nao_Pula } from "./Nao_Pula";

export class Pato_Borracha extends Pato {
    constructor() {
        super();
        this.setComportamento(new Nao_Voa(), new Nao_Pula());
    }

    mostrar(): string {
        return "Eu sou um Pato de Borracha!";
    }
}
